add log function that uses the `log` crate

jaemk opened the pull request:
It'd be nice to have the option to incorporate request logging with an app's `env_logger` or whatever logging backend is being used.

The name is up for debate, not sure what else to call an alternate logging function.

tomaka commented:
I'm not familiar enough with the `log` crate. Will we need to bump the major version of rouille if the log crate gets bumped to 0.4 in the future?

jaemk commented:
Ah, that's a good point. The "loggers" implement traits from the `log` crate, so yeah it seems you would need to keep the `log` crate up to date to work with the latest "loggers". That's unfortunate. Maybe the alternate logging function could instead return a tuple containing the formatted log message? Then it can be passed onto whatever logger is being used.

```rust
// still not sure about the name
fn request_info<F: FnOnce()> -> Response>(rq: &Request, f: F) -> (String, Response);
```

jaemk commented:
Scratch that. Returning a message wouldn't work when the handler panics

A better solution would probably be to accept logging functions:

```rust
fn log_custom<F, L, E>(rq: &Request, log_ok_f: L, log_err_f: E, handler: F) -> Response
    where F: FnOnce() -> Response,
          L: Fn(&Request, &Response, std::time::Duration),
          E: Fn(&Request, std::time::Duration)
```

jaemk commented:
Alright, I've updated to the `log_custom` signature above. Also updated `log` docs and bumped `chrono` to 0.4.0 (only used internally by `rouille::log`).

tomaka commented on src/log.rs:
```diff
@@ -63,6 +67,65 @@ pub fn log<W, F>(rq: &Request, mut output: W, f: F) -> Response
+///
+/// fn handle(request: &Request) -> Response {
+///     let now = chrono::Utc::now().format("%Y-%m-%d %H:%M:%S%.6f");
+///     let log_err = |req: &Request, _elap: std::time::Duration| {
```
Should be next to `log_ok`.

jaemk replied:
I've moved `log_ok` down next to `log_err`. Wanted to make a point that you need to capture the current time to get the proper timestamp of the request since you can't be sure when the log functions will run (unless you do datetime math in the log function).
